### DISCUSSION TITLE
What methods are currently available to export geological models generated by Gempy?

### DISCUSSION THREAD

**Discussion by FisPrObXu** [gempy-project/gempy#916] [General]
I am currently learning GemPy and also exploring software like FLAC3D. I want to export geological models generated by GemPy so that they can not only be previewed in software like Blender but also imported into FLAC3D for subsequent discretization and meshing. However, I can only find different methods for exporting models from various forums and platforms, some of which are no longer applicable to GemPyV3.

I would like to know what model export methods are currently available. If possible, I would appreciate some examples or tutorials. 
Thank you!

**Comment by javoha** [comment]
Hello @FisPrObXu,
basically all the part of a model are stored as individual arrays for further use. I will try to concentrate on the most import ones:

For visualization in Blender you probably want the 3D Dual Contouring meshes (those are the ones you also see in the 3D plots). The vertices and edges are stored here (index refers to structural element):
```
geo_model.solutions.dc_meshes[0].vertices
geo_model.solutions.dc_meshes[0].edges
```
Note that the xyz coordinates of the vertices will be in a transformed state. To get the real coordinates use:
`geo_model.input_transform.apply_inverse(geo_model.solutions.dc_meshes[0].vertices)`

The coordinates of your grid are stored in:
`geo_model.grid.regular_grid.values`

And the corresponding lithology ID values are in:
`geo_model.solutions.raw_arrays.lith_block`

You can find more parts of the resulting model in `geo:model.solutions.raw_arrays`, but I believe these are the most important/useful ones.

Cheers,
Jan

**Reply by FisPrObXu** [reply]
Thank you so much, this is really helpful.
